evmap-derive/src/lib.rs

```diff
@@ -61,9 +61,21 @@ fn fields(data: &Data, type_name: &Ident) -> TokenStream {
         }
         Data::Enum(data) => {
             let recurse = data.variants.iter().map(|f| {
+                let field_names = f.fields.iter().enumerate().map(|(i, field)| {
+                    let ident = format_ident!("x{}", i);
+                    quote_spanned! {
+                        field.span()=> #ident
+                    }
+                });
+                let fields = f.fields.iter().enumerate().map(|(i, field)| {
+                    let ident = format_ident!("x{}", i);
+                    quote_spanned! {field.span()=>
+                       std::mem::ManuallyDrop::into_inner(evmap::shallow_copy::ShallowCopy::shallow_copy(#ident))
+                    }
+                });
                 let name = &f.ident;
                 quote_spanned! {f.span()=>
-                    #type_name::#name(x) => std::mem::ManuallyDrop::new(#type_name::#name(std::mem::ManuallyDrop::into_inner(evmap::shallow_copy::ShallowCopy::shallow_copy(x))))
+                    #type_name::#name(#(#field_names,)*) => std::mem::ManuallyDrop::new(#type_name::#name(#(#fields,)*))
```

Comment:
This still seems like it would only support tuple-style enum variants? Having some tests here would be very good.